tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,435 @@
+		hubIPv4Network           = "172.16.0.0/24"
+		masterPodIPv4Address     = "172.16.0.1"
+		portNum8888              = 8888
+		cnfWorkerNodeList        []*nodes.Builder
+		masterNodeList           []*nodes.Builder
+		ipv4NodeAddrList         []string
+		ip4Worker0NodeAddr       []string
+		ipv4SecurityIPList       []string
+		testPodWorker0           *pod.Builder
+		testPodList              []*pod.Builder
+		routeMap                 map[string]string
+		mcNftablesName           = "98-nftables-cnf-worker"
+		interfaceNameNet1        = "net1"
+		interfaceNameBrEx        = "br-ex"
+		err                      error
+	)
+	BeforeAll(func() {
+		By("List CNF worker nodes in cluster")
+		cnfWorkerNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+		By("Selecting worker node for Security tests")
+		ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+			APIClient, metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+		ip4Worker0NodeAddr = []string{ipv4NodeAddrList[0]}
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(cnfWorkerNodeList)).To(BeNumerically(">", 1),
+			"Failed to detect at least two worker nodes")
+
+		By("verify status of nftables and activate if inactive")
+		activateNftablesIfInactive()
+		verifyNftablesStatus("active", cnfWorkerNodeList[0].Definition.Name)
+
+		By("Edit the machineconfiguration cluster to include NFTables")
+		updateMachineConfigurationNodeDisruptionPolicy()
+
+		By("Create test pods on worker node 0")
+		testPodWorker0 = createTestPodOnWorkers(cnfWorkerNodeList[0].Definition.Name, portNum8888)
+		testPodList = []*pod.Builder{testPodWorker0}
+
+		By("Create a static route to the external Pod network on each worker node")
+		// GetMetalLbVirIP using the metallb virtal IP address variable for test pod IP addresses.
+		ipv4SecurityIPList, err = NetConfig.GetMetalLbVirIP()
+		Expect(err).ToNot(HaveOccurred(), "Failed to retrieve the ipv4SecurityIPList")
+
+		routeMap, err = netenv.BuildRoutesMapWithSpecificRoutes(testPodList, cnfWorkerNodeList, ipv4SecurityIPList)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create route map with specific routes")
+
+		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "add", hubIPv4Network)
+	})
+
+	AfterAll(func() {
+		By("Remove the static route to the external Pod network on each worker node")
+		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "del", hubIPv4Network)
+
+		By("Disables nftables in cluster")
+		disableNftablesIfActive()
+		verifyNftablesStatus("inactive", cnfWorkerNodeList[0].Definition.Name)
```

Comment:
Please move it to inside the func disableNftablesIfActive())

Reply:
moved